### ISSUE TITLE
"ext_id" on the VOD Information always a mismatch.

### ISSUE THREAD

**Issue by Zibbp** [Zibbp/ganymede#346]
### Discussed in https://github.com/Zibbp/ganymede/discussions/345

<div type='discussions-op-text'>

<sup>Originally posted by **CappiSteijns** January  7, 2024</sup>
Wouldn't count this as an issue because beside archiving and watching on Ganymede I'm using the VOD Information from ganymede to make an index on my website so other people can easily find vods trough playlists I made in Ganymede.

For example: https://lekkerspelenfandom.nl/untitled-2/

But take this VOD information as example:

```
{
  "id": "85f6251d-99f0-11ee-9e8f-0242ac140003",
  "ext_id": "42114447897",
  "platform": "twitch",
  "type": "live",
  "title": "PETER VS TIMON 2023 🏆",
  "duration": 9534,
  "views": 1,
  "resolution": "best",
  "thumbnail_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-thumbnail.jpg",
  "web_thumbnail_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-web_thumbnail.jpg",
  "video_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-video.mp4",
  "chat_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-chat.json",
  "chat_video_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-chat.mp4",
  "info_path": "/vods/lekkerspelen/2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003/42114447897-info.json",
  "folder_name": "2023-12-13-42114447897-live-85f6251d-99f0-11ee-9e8f-0242ac140003",
  "file_name": "42114447897",
  "local_views": 1,
  "streamed_at": "2023-12-13T19:48:01.377099Z",
  "updated_at": "2024-01-07T03:13:33.711539Z",
  "created_at": "2023-12-13T19:48:01.377256Z",
  "edges": {
    "channel": {
      "id": "da865789-421d-445f-8fc2-817fd2446111",
      "ext_id": "52385053",
      "name": "lekkerspelen",
      "display_name": "lekkerspelen",
      "image_path": "/vods/lekkerspelen/profile.png",
      "updated_at": "2023-02-01T10:50:34.954278Z",
      "created_at": "2023-01-22T15:19:14.945367Z",
      "edges": {}
    }
  }
}
```
You would think the "ext_id" would be the id of the vod? So that would be https://www.twitch.tv/videos/42114447897.
But that doesn't exist, it actually is in this case: https://www.twitch.tv/videos/2003234717.

So I was thinking that maybe twitch changes the "ext_id" when it's going from live to vod?

Probably out of scope for this project but was wondering if there was an fix for this. </div>

**Comment by Zibbp** [comment]
Twitch has a VOD ID and a live ID which is why they are different. Once the video download is complete, I can attempt to update the Ganymede VOD with the Twitch VOD ID that corresponds with the live stream. This obviously has some catches. The VOD needs to be available immediately so the channel needs to not have the manual vod creation setting enabled (where the channel needs to click a button to publish the vod), and needs vods turned on in general.

If the Ganymede livestream video does get updated with the correct external ID of the VOD then you won't be able to archive the VOD as it technically already exists in Ganymede (the external IDs are the same). Probably not a huge concern as you already have the livestream.

Would this solve your issue?

**Comment by CappiSteijns** [comment]
> Would this solve your issue?

Yes, sounds good!

>If the Ganymede livestream video does get updated with the correct external ID of the VOD then you won't be able to archive the VOD as it technically already exists in Ganymede

For me what won't be a problem at all, as you said, I already have the VOD.

Thank you!

**Comment by Zibbp** [comment]
Released in https://github.com/Zibbp/ganymede/releases/tag/v2.0.2

**Comment by CappiSteijns** [comment]
> Released in https://github.com/Zibbp/ganymede/releases/tag/v2.0.2

Thank you! Any way I can force this on existing vods?

**Comment by Zibbp** [comment]
I've added a task that runs the logic for all channels in https://github.com/Zibbp/ganymede/pull/351. It is available using the `:main` image tag for both the frontend and API images. To execute this, visit Admin > Tasks.

![image](https://github.com/Zibbp/ganymede/assets/21207065/1ced5460-6015-44a5-9e1e-37b2635597ae)

**Comment by CappiSteijns** [comment]
> I've added a task that runs the logic for all channels in #351. It is available using the `:main` image tag for both the frontend and API images. To execute this, visit Admin > Tasks.
> 
> ![image](https://private-user-images.githubusercontent.com/21207065/295092608-1ced5460-6015-44a5-9e1e-37b2635597ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MDIxNDYsIm5iZiI6MTcwNDgwMTg0NiwicGF0aCI6Ii8yMTIwNzA2NS8yOTUwOTI2MDgtMWNlZDU0NjAtNjAxNS00NGE1LTllMWUtMzdiMjYzNTU5N2FlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA5VDEyMDQwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkZjJhZjdlMTU4MmU2ZjA4YzNlYjVkMDU2MDgzYzc1M2RjZGRiZmJhODgzNjIzZGYxYWVlYzZjNGFlYWU2MTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vvEeoxAE5bFAXsy60QI8mt6D9_WsIIHpSAU7kmtpZ-k)

Thank you so much! Seems to be working :D

**Comment by CappiSteijns** [comment]
> > I've added a task that runs the logic for all channels in #351. It is available using the `:main` image tag for both the frontend and API images. To execute this, visit Admin > Tasks.
> > ![image](https://private-user-images.githubusercontent.com/21207065/295092608-1ced5460-6015-44a5-9e1e-37b2635597ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MDIxNDYsIm5iZiI6MTcwNDgwMTg0NiwicGF0aCI6Ii8yMTIwNzA2NS8yOTUwOTI2MDgtMWNlZDU0NjAtNjAxNS00NGE1LTllMWUtMzdiMjYzNTU5N2FlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA5VDEyMDQwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkZjJhZjdlMTU4MmU2ZjA4YzNlYjVkMDU2MDgzYzc1M2RjZGRiZmJhODgzNjIzZGYxYWVlYzZjNGFlYWU2MTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vvEeoxAE5bFAXsy60QI8mt6D9_WsIIHpSAU7kmtpZ-k)
> 
> Thank you so much! Seems to be working :D

After a little tinkering with it I found out it only works for the latest ~18 vod's located at the =archives (example: https://www.twitch.tv/lekkerspelen/videos?filter=archives)  on a twitch channel. 
While all the other vods are located at =all (example https://www.twitch.tv/lekkerspelen/videos?filter=all&sort=time)

Now I noticed that those also have a different vod id...

It does work for the newly added vod's automatically tho so thats good! :)

So would it be possible to have the older live archives also corrected with the right vod id?

**Comment by Zibbp** [comment]
The API call to Twitch to fetch videos is scoped to only return `archive` videos https://github.com/Zibbp/ganymede/blob/main/internal/activities/video.go#L1096. My logic behind that is only livestreams would need their ID updated so the only video type that should be searched are archives right? Any video other than an archive likely doesn't have the `livestream_id` attribute on it that I use to match the live stream to the VOD.
